task3.py

```diff
@@ -0,0 +1,25 @@
+from pyspark.sql import functions as f
+from pyspark.sql.window import Window
+
+from utilities import window as g_window
+
+
+def decade_top_films(df):
+    """
+    Find top 10 films in every genre for decades since 1950
+    """
+
+    d_window = Window.partitionBy('decades') \
+        .orderBy(f.col('decades').desc())
+
+    df = df \
+        .withColumn('genres', f.explode(f.split('genres', ','))) \
```

Comment:
you have already done this in task2, it's better to do it only once and then use df with exploded `genres` in task2 and in task3